src/lib/y2s390/zfcp.rb

```diff
@@ -0,0 +1,183 @@
+
+    # Detected controllers
+    #
+    # @return [Array<Hash>] keys for each hash:
+    #   "sysfs_bus_id", "resource"
+    attr_reader :controllers
+
+    # Detected LUN disks
+    #
+    # @return [Array<Hash>] keys for each hash:
+    #   "dev_name", "detail", "vendor", "device" and "io".
+    attr_reader :disks
+
+    def initialize
+      @controllers = []
+      @disks = []
+    end
+
+    # Probes the zFCP controllers
+    def probe_controllers
+      make_all_devices_visible
+
+      storage_devices = Yast::SCR.Read(path(".probe.storage"))
+      controllers = storage_devices.select { |i| i["device"] == "zFCP controller" }
+
+      @controllers = controllers.map { |c| c.slice("sysfs_bus_id", "resource") }
+    end
+
+    # Probes the zFCP disks
+    def probe_disks
+      storage_disks = read_mock_disks || Yast::SCR.Read(path(".probe.disk"))
+      zfcp_disks = storage_disks.select { |d| d["driver"] == "zfcp" }
+
+      tapes = Yast::SCR.Read(path(".probe.tape"))
+      scsi_tapes = tapes.select { |t| t["bus"] == "SCSI" }
+
+      disks = zfcp_disks + scsi_tapes
+
+      @disks = disks.map { |d| d.slice("dev_name", "detail", "vendor", "device", "io") }
+    end
+
+    # Runs the command for activating a controller
+    #
+    # @note All LUNs are automatically activated if "allow_lun_scan" is active, see
+    #   https://www.ibm.com/docs/en/linux-on-systems?topic=wsd-configuring-devices.
+    #
+    # @param channel [String] E.g., "0.0.fa00"
+    # @return [Hash] See {#run}. The exit code corresponds to the chzdev one.
+    def activate_controller(channel)
+      command = format("/sbin/zfcp_host_configure '%s' %d", channel, 1)
+      run(command)
+    end
+
+    # Whether the controller is activated
+    #
+    # @param channel [String] E.g., "0.0.fa00"
+    # @return [Boolean]
+    def activated_controller?(channel)
+      controller = controllers.find { |c| c["sysfs_bus_id"] == channel }
+      return false unless controller
+
+      io = controller.dig("resource", "io") || []
+      io.any? { |i| i["active"] }
+    end
+
+    # Runs the command for activating a zFCP disk
+    #
+    # @param channel [String] E.g., "0.0.fa00"
+    # @param wwpn [String] E.g., "0x500507630708d3b3"
+    # @param lun [String] E.g., "0x0013000000000000"
+    #
+    # @return [Hash] See {#run}. The exit code corresponds to the chzdev one.
+    def activate_disk(channel, wwpn, lun)
+      command = format("/sbin/zfcp_disk_configure '%s' '%s' '%s' %d", channel, wwpn, lun, 1)
+      run(command)
+    end
+
+    # Runs the command for deactivating a zFCP disk
+    #
+    # @note Deactivate fails if "allow_lun_scan" is active, see
+    #   https://www.ibm.com/docs/en/linux-on-systems?topic=wsd-configuring-devices.
+    #
+    # @param channel [String] E.g., "0.0.fa00"
+    # @param wwpn [String] E.g., "0x500507630708d3b3"
+    # @param lun [String] E.g., "0x0013000000000000"
+    #
+    # @return [Hash] See {#run}. The exit code corresponds to the chzdev one.
```

Comment:
Sorry, asking from ignorance. You've written "The exit code corresponds to the chzdev one" in three different methods here. Is it ok or a copy/paste leftover? Thanks for clarifying it.